pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -626,6 +626,11 @@ func (c *Reconciler) makeConditionCheckContainer(rprt *resources.ResolvedPipelin
 	labels := getTaskrunLabels(pr, rprt.PipelineTask.Name)
 	labels[pipeline.GroupName+pipeline.ConditionCheckKey] = rcc.ConditionCheckName
 
+	taskSpec, err := rcc.ConditionToTaskSpec()
+	if err != nil {
+		return nil, err
```

Comment:
its often helpful to augment the error with some context before returning it, e.g. ` xerrors.Errorf("Failed to get TaskSpec from condition: %w", err)`